core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -101,25 +98,47 @@ public final boolean isConfigured() {
      * the category has been successfully configured. Whereas configuration is a short-lived
      * synchronous phase of setup, initialization may require disk/network resources, etc.
      * @param context An Android Context
-     * @param onInitializationAttempted Called when initialization has been attempted.
-     *                                  The result contains information about each plugin,
-     *                                  and whether or not its initialization succeeded.
+     * @return A category initialization result
      */
-    public final synchronized void initialize(
-            @NonNull Context context,
-            @NonNull Consumer<CategoryInitializationResult> onInitializationAttempted) {
-        Map<String, InitializationResult> pluginInitializationResults = new HashMap<>();
-        for (P plugin : getPlugins()) {
-            InitializationResult result;
-            try {
-                plugin.initialize(context);
-                result = InitializationResult.success();
-            } catch (AmplifyException pluginInitializationFailure) {
-                result = InitializationResult.failure(pluginInitializationFailure);
+    @NonNull
+    @WorkerThread
+    public final synchronized CategoryInitializationResult initialize(@NonNull Context context) {
+        final Map<String, InitializationResult> pluginInitializationResults = new HashMap<>();
+        if (!State.CONFIGURED.equals(state.get())) {
```

Comment:
I really like the new `State` enum here but I will admit that readability of the if-not-equals-get is a bit much and someone could introduce a bug in the future if they're not careful. Not sure I have an answer just calling it out.

Reply:
The enum is a decent way to handle the state machine, however, I would have liked to avoid the state machine. Hopefully this can be revisited ...